.scalafmt.conf

```diff
@@ -1,45 +1,140 @@
+  tokens."+" = [{
+    code = ":"
+    owners = [{
+      parents = ["Defn\\."]
+    },
+      {
+        regex = "Term\\.Param"
+        parents = ["Ctor\\.Primary"]
+      }]
+  }, {
+    code = "=>"
+    owners = [{
+      regex = "Case"
+    }]
+  }
+  ]
+}
 
-align = more
-align.openParenCallSite = false
-align.openParenDefnSite = false
-align.tokens = [
-  {code = "=>", owner = "Case"},
-  {code = "%", owner = "Term.ApplyInfix"},
-  {code = "%%", owner = "Term.ApplyInfix"}  
-]
+newlines {
+  source = keep
+  alwaysBeforeElseAfterCurlyIf = false
+  beforeMultiline = fold
+  afterCurlyLambdaParams = squash
+  avoidForSimpleOverflow = [slc, tooLong, punct] # singe line comment, does not add newline if line would be too long after,  punctuation
+  forceBeforeMultilineAssign = never
+}
 
-newlines.alwaysBeforeTopLevelStatements = false
-newlines.alwaysBeforeElseAfterCurlyIf = false
-newlines.afterCurlyLambda = never
+verticalMultiline {
+  atDefnSite = true
+  arityThreshold = 6
+}
 
-rewrite.rules = [ExpandImportSelectors]
+# TODO Comments newline
+# alignment constructor
 
-rewrite.rules = [RedundantParens]
 
-rewrite.rules = [PreferCurlyFors]
+indent {
+  main = 4
+  callSite = 4
+  ctorSite = 8 # Double Indent
+  withSiteRelativeToExtends = 4
+}
+
+assumeStandardLibraryStripMargin = true
+lineEndings = unix
+binPack.parentConstructors = keep
+includeNoParensInSelectChains = false
+optIn.breakChainOnFirstMethodDot = false
+optIn.configStyleArguments = true
+optIn.breaksInsideChains = false
+rewrite {
+  rules = [
+    RedundantParens,
+    RedundantBraces,
+    SortModifiers,
+    PreferCurlyFors,
+    Imports
+  ]
+  sortModifiers.order = [
+    "implicit", "final", "sealed", "abstract",
+    "override", "private", "protected", "lazy"]
+  redundantBraces.generalExpressions = false
+  redundantBraces.parensForOneLineApply = false
+  preferCurlyFors.removeTrailingSemicolonsOnly = false
+  imports {
+    expand = true
+    sort = scalastyle
+    groups = [
+      ["scala\\.language\\..*",
+        "scala\\.annotation\\..*",
+        "scala\\.reflect\\..*"],
+      ["javax?\\..*",
+        "scala\\..*",
+        "play\\..*"],
+      ["org\\.opalj\\..*"],
+      ["scala\\.collection\\.parallel\\.CollectionConverters\\..*"]
+      // Catch all
+    ]
+  }
+}
 
-rewrite.rules = [sortmodifiers]
-rewrite.sortModifiers.order = [
-  "implicit", "final", "sealed", "abstract",
-  "override", "private", "protected", "lazy"
+literals {
+  long = Upper
+  float = Lower
+  double = Lower
+}
+
+
+# There is something *really* broken with the report strings not even // format: off works
+project.excludePaths = [
+  "glob:**/DEVELOPING_OPAL/demos/src/main/scala/org/opalj/fpcf/analyses/InterProceduralEscapeAnalysisDemo.scala",
+  "glob:**/DEVELOPING_OPAL/demos/src/main/scala/org/opalj/fpcf/analyses/SimpleEscapeAnalysisDemo.scala"
 ]
 
-literals.long = Upper
-literals.float = Lower
-literals.double = Lower
-lineEndings = unix
+fileOverride {
```

Comment:
What is the point of this? At least document if necessary.

Reply:
We can remove that since we exclude the files explicitly now